crates/mun_language_server/src/state/protocol/dispatcher.rs

```diff
@@ -41,6 +41,36 @@ impl<'a> RequestDispatcher<'a> {
         Ok(self)
     }
 
+    /// Try to dispatch the event as the given Request type on the thread pool.
+    pub fn on<R>(
```

Comment:
```suggestion
    pub fn on_request<R>(
```

Reply:
It seems like the subject of "on" is missing. Is the subject the request?

Reply:
The subject is `R`. It's written like: `.on::<lsp_types::request::DocumentSymbolRequest>(handlers::handle_document_symbol)?`

Reply:
Either works